clover/netcdf/variable.py

```diff
@@ -464,6 +464,35 @@ def datetimes(self):
         else:
             return numpy.array([datetime(*d.timetuple()[:6], tzinfo=pytz.UTC) for d in self.dates])
 
+    @property
+    def unit(self):
+        def varys_by_year(x, y):
+            if y.year == x.year or (y - x).seconds != 0 or x.month != y.month or x.day != y.day:

```

Comment:
That's not quite the same. The original statement is essentially saying: "if there is no difference in year, or if there _are_ differences in seconds, months, or days, then these dates don't vary by year."